src/unstable-temp/DownloadImageModal/OsConfiguration.tsx

```diff
@@ -289,13 +314,23 @@ export const OsConfiguration = ({
 								valueKey="value"
 								labelKey="title"
 								emptySearchMessage="No version available for this application type"
-								value={version ?? {}}
+								value={
+									version ? (
```

Comment:
I think that this is not supposed to be  JSX element, or more accurately, be one of the items provided in the `options` prop.
The JSX of the current selected value goes to the `valueLabel` prop afaict.